src/Serializers/Nest.JsonNetSerializer/Converters/HandleNestTypesOnSourceJsonConverter.cs

```diff
@@ -40,14 +41,19 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 				return _builtInSerializer.Deserialize(objectType, ms);
 		}
 
-		private static readonly Type[] NestTypesThatCanAppearInSource = {
+		private static readonly HashSet<Type> NestTypesThatCanAppearInSource = new HashSet<Type>
+		{
 			typeof(JoinField),
 			typeof(QueryContainer),
 			typeof(CompletionField),
 			typeof(Attachment),
-			typeof(ILazyDocument)
+			typeof(ILazyDocument),
+			typeof(GeoCoordinate)
```

Comment:
`IGeoShape` as well as `IGeometryCollection` ?

Reply:
Handled on lines 56 and 57 below. It's possible that a user could specify any type that implements `IGeoShape` e.g. `IPolygonGeoShape`, so json conversion should check if `IGeoShape` is assignable from the type.